bluesky/bundlers.py

```diff
@@ -274,37 +341,11 @@ async def monitor(self, msg):
                 "A 'monitor' message was sent for {}"
                 "which is already monitored".format(obj)
             )
-        descriptor_uid = new_uid()
-        if obj not in self._describe_cache:
-            await self._cache_describe(obj)
-        if obj not in self._config_desc_cache:
-            await self._cache_describe_config(obj)
-            await self._cache_read_config(obj)
-        data_keys = self._describe_cache[obj]
-        config = {obj.name: {
-            "data": self._config_values_cache[obj],
-            "timestamps": self._config_ts_cache[obj],
-            "data_keys": self._config_desc_cache[obj]
-        }}
-        object_keys = {obj.name: list(data_keys)}
-        hints = {}
-        maybe_update_hints(hints, obj)
-        desc_doc = dict(
-            run_start=self._run_start_uid,
-            time=ttime.time(),
-            data_keys=data_keys,
-            uid=descriptor_uid,
-            configuration=config,
-            hints=hints,
-            name=name,
-            object_keys=object_keys,
-        )
-        doc_logger.debug("[descriptor] document is emitted with name %r containing "
-                         "data keys %r (run_uid=%r)", name, data_keys.keys(),
-                         self._run_start_uid,
-                         extra={'doc_name': 'descriptor',
-                                'run_uid': self._run_start_uid,
-                                'data_keys': data_keys.keys()})
+        await self._cache_describe(obj)
```

Comment:
The describe cache from `read()` was previously used here:
```suggestion
        if obj not in self._describe_cache:
            await self._cache_describe(obj)
```
Similar for the `read_configuration()` and `describe_configuration()`. Is this refresh of the cache on monitor purposeful?

Reply:
Fixed this and centralized most of this into one place.

Could not fully inline the helpers because collect also uses some of them.  Will leave that refactor for the future.